cuml/rf_demo.ipynb

```diff
@@ -0,0 +1,439 @@
+       "3      0  1.344385 -0.876626  0.935913  1.992050  0.882454  1.786066   \n",
+       "4      1  1.105009  0.321356  1.522401  0.882808 -1.205349  0.681466   \n",
+       "\n",
+       "      col-8     col-9    col-10  ...    col-20    col-21    col-22    col-23  \\\n",
+       "0 -0.248573 -1.092064  0.000000  ... -0.010455 -0.045767  3.101961  1.353760   \n",
+       "1 -0.557525 -1.588230  2.173076  ... -1.138930 -0.000819  0.000000  0.302220   \n",
+       "2  1.282323  1.381664  0.000000  ...  1.128848  0.900461  0.000000  0.909753   \n",
+       "3 -1.646778 -0.942383  0.000000  ... -0.678379 -1.360356  0.000000  0.946652   \n",
+       "4 -1.070464 -0.921871  0.000000  ... -0.373566  0.113041  0.000000  0.755856   \n",
+       "\n",
+       "     col-24    col-25    col-26    col-27    col-28    col-29  \n",
+       "0  0.979563  0.978076  0.920005  0.721657  0.988751  0.876678  \n",
+       "1  0.833048  0.985700  0.978098  0.779732  0.992356  0.798343  \n",
+       "2  1.108331  0.985692  0.951331  0.803251  0.865924  0.780118  \n",
+       "3  1.028704  0.998656  0.728281  0.869200  1.026736  0.957904  \n",
+       "4  1.361057  0.986610  0.838085  1.133295  0.872245  0.808486  \n",
+       "\n",
+       "[5 rows x 29 columns]"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Read Higgs data using cudf\n",
+    "\n",
+    "compressed_filepath = '/datasets/higgs/HIGGS.csv.gz'\n",
+    "decompressed_filepath = '/datasets/higgs/HIGGS.csv'\n",
+    "download_higgs(compressed_filepath, decompressed_filepath)\n",
+    "\n",
+    "col_names = ['label'] + [\"col-{}\".format(i) for i in range(2, 30)]\n",
+    "dtypes_ls = ['int32'] + ['float32' for _ in range(2, 30)]\n",
+    "data = cudf.read_csv(decompressed_filepath, names=col_names, dtype=dtypes_ls)\n",
+    "data.head().to_pandas()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Seperate into X and y\n",
+    "\n",
+    "X, y = data[data.columns.difference(['label'])].as_matrix(), data['label'].to_array()\n",
+    "del data\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=500_000)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(10500000, 28) (10500000,) (500000, 28) (500000,)\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(X_train.shape, y_train.shape, X_test.shape, y_test.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# cuml Random Forest params\n",
+    "\n",
+    "cu_rf_params = {\n",
+    "    'n_estimators': 25,\n",
+    "    'max_depth': 13,\n",
+    "    'n_bins': 15,\n",
```

Comment:
cuML's implementation is a little bit different than SKL's implementation. For instance, SKL doesn't have n_bins parameter. Would be nice to add some comments about this different implementation and the different parameters.